cypress/integration/internal/license-transfer-to-new-billing-account.spec.js

```diff
@@ -0,0 +1,236 @@
+      cy.get('.govuk-heading-l').contains('Add a new contact');
+      describe('error - empty', () => {
+        cy.get('form > .govuk-button').contains('Continue').click();
+        checkInlineAndSummaryErrorMessage('Enter a first name');
+        cy.reload();
+      });
+      describe('error - no last name  name', () => {
+        cy.get('#firstName').type(firstName);
+        cy.get('form > .govuk-button').contains('Continue').click();
+        checkInlineAndSummaryErrorMessage('Enter a last name');
+        cy.reload();
+      });
+
+      describe('good', () => {
+        cy.get('#firstName').type(firstName);
+        cy.get('#lastName').type(lastName);
+        cy.get('form > .govuk-button').contains('Continue').click();
+      });
+    });
+
+    describe('Check billing account details', () => {
+      cy.get('.govuk-heading-l').contains('Check billing account details');
+
+      cy.get('.govuk-summary-list__row').should(items => {
+        expect(items[0]).to.contain.text('Billing contact name');
+        expect(items[0]).to.contain.text(fullName);
+        expect(items[1]).to.contain.text('Billing address');
+        expect(items[1]).to.contain.text('\n  \n    ' +
+            'Environment Agency\n  \n  \n    ' +
+            'Horizon House\n  \n    \n  ' +
+            'Deanery Road\n  \n  \n  \n    ' +
+            'Bristol\n  \n  \n  \n    ' +
+            'BS1 5AH\n  \n  \n    ' +
+            'United Kingdom\n');
+        expect(items[2]).to.contain.text('FAO');
+        expect(items[2]).to.contain.text(`${firstName}\n    ${lastName}`);
+      });
+      cy.get('button.govuk-button').click();
+    });
+
+    describe('Use abstraction data to set up the element?', () => {
+      validateRadioOptions('Use abstraction data to set up the element?', 'useAbstractionData', 'Select whether to use abstraction data to set up the element');
+    });
+
+    describe('Check charge information', () => {
+      cy.get('.govuk-heading-xl').contains('Check charge information');
+
+      cy.get('.govuk-summary-list__row').should(items => {
+        expect(items[0]).to.contain.text('Reason');
+        expect(items[0]).to.contain.text('Strategic review of charges (SRoC)');
+        expect(items[1]).to.contain.text('Start date');
+        expect(items[1]).to.contain.text('1 June 2022');
+        expect(items[2]).to.contain.text('Billing account');
+        expect(items[2]).to.contain.text(
+          'A00000001A\n              \n\n              \n              \n                ' +
+            'FAO\n                \n  \n    \n    ' +
+            'Jim\n    Bob\n  \n\n              \n              \n                \n  ' +
+            'John Smith\n\n              \n              \n  \n    ' +
+            'Environment Agency\n  \n  \n    ' +
+            'Horizon House\n  \n    \n  ' +
+            'Deanery Road\n  \n  \n  \n    ' +
+            'Bristol\n  \n  \n  \n    ' +
+            'BS1 5AH\n  \n  \n    ' +
+            'United Kingdom\n');
```

Comment:
This test is failing for me as the string is too specific -- I get a different reference to `A00000001A`. Since the data we're entering is the postcode, first name, last name and full name, we could maybe just confirm that these are all present in `items[2]`?

Reply:
I'd have expected the tests to pass with the same license as the setup step

Reply:
No issue removing the license ref but it would be good to understand why the data is different.

Reply:
```suggestion
        expect(items[2]).to.contain.text(
            'FAO\n                \n  \n    \n    ' +
            'Jim\n    Bob\n  \n\n              \n              \n                \n  ' +
            'John Smith\n\n              \n              \n  \n    ' +
            'Environment Agency\n  \n  \n    ' +
            'Horizon House\n  \n    \n  ' +
            'Deanery Road\n  \n  \n  \n    ' +
            'Bristol\n  \n  \n  \n    ' +
            'BS1 5AH\n  \n  \n    ' +
            'United Kingdom\n');
```